src/extension/features/budget/custom-average-budgeting/index.js

```diff
@@ -0,0 +1,111 @@
+import { controllerLookup, getEmberView } from 'toolkit/extension/utils/ember';
+import {
+  getEntityManager,
+  isCurrentMonthSelected,
+  isCurrentRouteBudgetPage,
+} from 'toolkit/extension/utils/ynab';
+import { formatCurrency } from 'toolkit/extension/utils/currency';
+import { Feature } from '../../feature';
+
+export class CustomAverageBudgeting extends Feature {
+  constructor() {
+    super();
+  }
+
+  shouldInvoke() {
+    return isCurrentRouteBudgetPage() && isCurrentMonthSelected();
+  }
+
+  _calculateLookback() {
+    const today = new Date();
+    let d;
+    let month;
+    let year;
+    const arr = [];
+
+    for (var i = this.settings.enabled; i >= 1; i -= 1) {
+      d = new Date(today.getFullYear(), today.getMonth() - i, 1);
+      month = d.getMonth() + 1;
+      year = d.getFullYear();
+      arr.push(`${month}|${year}`);
+    }
+
+    return arr.join('`');
+  }
+
+  _calculateAverage() {
+    const timeframe = this._calculateLookback().split('`');
+    console.log(timeframe);
+    let sum = 0;
+    getEntityManager().transactionsCollection._internalDataArray.forEach((transaction) => {
```

Comment:
there is a `forEach` on `transactionsCollection` so no need to access `_internalDataArray` -- in general the `_` means that thing is private and we should avoid using those especially when they're internal to YNAB/ember like this